include/nighthawk/common/poolable.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+namespace Nighthawk {
+
+class Poolable {
```

Comment:
As you are using templates, I don't think you need this ABC at all. It's nice in a sense as documentation and compiler enforcement that your implementations implement your interface, but it costs you some memory and performance without adding functionality in this design, AFAICT.

How many Poolables do you expect to have lying around during operation? Maybe the above doesn't matter.